server/files/api_views.py

```diff
@@ -53,3 +55,71 @@ def update(self, request, pk):
         return Response(FilesSerializer(file_obj_to_update).data, status=201)
 
     queryset = File.objects.all()
+
+
+class NotebookFileSourceViewSet(viewsets.ModelViewSet):
+
+    serializer_class = FileSourceDetailSerializer
+    http_method_names = ["get"]
+
```

Comment:
We should be able to use the `get_serializer_class` method here to pick the right serializer, depending on whether the user is logged in or not.